keylime/ima/file_signatures.py

```diff
@@ -184,21 +184,25 @@ def from_string(stringrepr: str) -> Optional["ImaKeyring"]:
         if not stringrepr:
             return None
 
-        ima_keyring = ImaKeyring()
-
-        default_be = backends.default_backend()
-
-        # An empty Db entry comes as a string '[]'. A valid DB entry as a string
-        # ith escaped quotes and needs to be loaded twice
+        # An empty Db entry comes as a string 'null' or previously '[]'. A valid DB entry
+        # as a string with escaped quotes and needs to be loaded twice
         obj = json.loads(stringrepr)
         if isinstance(obj, str):
             obj = json.loads(obj)
         if not isinstance(obj, dict):
             return None
 
-        keyids = obj.get("keyids", [])
+        return ImaKeyring.from_json(obj)
+
+    @staticmethod
+    def from_json(obj: Dict[str, Union[List[int], List[str]]]) -> "ImaKeyring":
```

Comment:
I think `TypedDict` should work here again. I'll try to finish the other PR, so that we can start using it.

Reply:
Let me pop this patch out and bring it again elsewhere later on. This split-off has time.